src/lib.rs

```diff
@@ -87,6 +87,33 @@
 //! cargo_metadata::metadata_run(Some(manifest_path), false, Some(features)).unwrap();
 //!
 //! ```
+//!
+//! Parse message-format output:
+//!
+//! ```
+//! # extern crate cargo_metadata;
+//! use std::process::{Stdio, Command};
+//! use cargo_metadata::Message;
+//! let command = Command::new("cargo")
+//!     .args(&["build", "--message-format=json"])
+//!     .stdout(Stdio::piped())
+//!     .spawn().unwrap();
+//!
+//! for message in cargo_metadata::parse_message_stream(command.stdout.unwrap()) {
```

Comment:
Imo this example should `.take()` before unwrapping. Otherwise you can't wait for the command to finish afterwards.